### DISCUSSION TITLE
Feature Request: UDF support for C API

### DISCUSSION THREAD

**Discussion by ayman-albaz** [duckdb/duckdb#4874] [General]
Hello DuckDB team.

One feature that I think might be useful is to bring UDF support for C API. Currently UDF functionality is supported via C++ API.

Bringing this feature to C would allow languages with C-based duckdb wrappers like Rust, Julia, Golang, and Nim to use this feature as well.

**Comment by Mytherin** [comment]
This is on our roadmap.

Scalar function support has been recently added to the C API in https://github.com/duckdb/duckdb/pull/11786
Table function/scanner support has been in the C API for a long time already.

What is missing currently are aggregate functions.

**Reply by Giorgi** [reply]
Is defining a scalar function without any input parameter supported? For example, I tried to define `my_rand` that would work like this: `SELECT my_rand() FROM sometable` but in the callback function I get a data chunk with size 1. If I add at least one parameter, I get the correct size.

**Comment by hannes** [comment]
This is available now
